src/builder.rs

```diff
@@ -617,28 +617,39 @@ impl ContainerOptionsBuilder {
         ContainerOptionsBuilder { name: None, params }
     }
 
-    pub fn name(
+    pub fn name<S>(
         &mut self,
-        name: &str,
-    ) -> &mut Self {
-        self.name = Some(name.to_owned());
+        name: S,
+    ) -> &mut Self
+    where
+        S: Into<String>,
```

Comment:
Well this method is different than others because the `name` it takes is actually later used as a `String` on contrary to other methods that take a `AsRef<str>` and us this argument as `&str`. So I think it actually makes more sense like this. But if i didn't convince you then sure enough I can make all of them `AsRef<str>` and then later call `to_string`